### PR TITLE
feat: 🎸 Set up route with gcloud store for list studies

### LINKED ISSUE #1056: deep link to study list for a dicom store
## Request

**What feature or change would you like to see made?**

Now that we can deep link to a study by putting DICOMweb server info in the URL, it would also make sense to be able to link to the particular store itself and get to the study list, like this:

https://idc-sandbox-000.firebaseapp.com/projects/idc-sandbox-000/locations/us-central1/datasets/sdp-dev/dicomStores/scratch
...

**Why should we prioritize this feature?**

It should be a good IDC feature.
...

### PR DESCRIPTION
**Pull request by ladeirarodolfo**
### Includes
- New route logic for gcloud list studies (deep link for store)

### Closes
#1056

### CODE REVIEW

File: platform/viewer/src/customHooks/useServer.js [hunk 1]
```diff
@@ -32,11 +32,22 @@ const getServers = (appConfig, project, location, dataset, dicomStore) => {
...
+    }
   }
 
   return servers;
 };
 
+const isValidServer = (server, appConfig) => {
+  if (appConfig.enableGoogleCloudAdapter) {
+    return GoogleCloudUtilServers.isValidServer(server);
+  } else {
```
Comment (swederik) [inline platform/viewer/src/customHooks/useServer.js]: I would have espected eslint to flag this: https://eslint.org/docs/rules/no-else-return

File: platform/core/src/utils/urlUtil.js [hunk 2]
```diff
@@ -55,11 +55,17 @@ const replaceParam = (path = '', paramKey, paramValue) => {
...
+  const paramPatternPiece = `/${PARAM_PATTERN_IDENTIFIER}`;
+  return path.indexOf(paramPatternPiece) < 0;
+};
+
 const queryString = {
   getQueryFilters: getFilters,
 };
 
 const paramString = {
+  isValidPath: isValidPath,
```
Comment (swederik) [inline platform/core/src/utils/urlUtil.js]: Same with this: https://eslint.org/docs/rules/object-shorthand

File: platform/viewer/src/customHooks/useServer.js (hunk 1 above)
Comment (ladeirarodolfo) [reply platform/viewer/src/customHooks/useServer.js]: ok

File: platform/core/src/utils/urlUtil.js (hunk 2 above)
Comment (ladeirarodolfo) [reply platform/core/src/utils/urlUtil.js]: ok

File: platform/viewer/CHANGELOG.md [hunk 3]
```diff
@@ -3,17 +3,14 @@
 All notable changes to this project will be documented in this file.
 See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.
 
+
```
Comment (swederik) [inline platform/viewer/CHANGELOG.md]: Why is your PR even changing this file? Remove this.

File: platform/viewer/src/customHooks/useServer.js [hunk 4]
```diff
@@ -32,28 +35,68 @@ const getServers = (appConfig, project, location, dataset, dicomStore) => {
...
+  if (!urlBasedServers || !urlBasedServers.length) {
+    return false;
+  } else if (!servers.length || !activeServer) {
+    // no current valid server
+    return true;
   }
+
+  const newServer = urlBasedServers[0];
+
+  let exists = _.some(servers, server => {
```
Comment (swederik) [inline platform/viewer/src/customHooks/useServer.js]: es6 Arrays can do this already, so you can remove Lodash for this: https://youmightnotneed.com/lodash/#some

**Pull request merged**